pkgs/development/python-modules/altair/default.nix

```diff
@@ -1,62 +1,70 @@
-{ lib, buildPythonPackage, fetchPypi, isPy27
-, entrypoints
-, glibcLocales
-, ipython
-, jinja2
-, jsonschema
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+
+# Run dependencies
+, hatchling
+, toolz
 , numpy
+, jsonschema
+, typing-extensions
 , pandas
+, jinja2
+, importlib-metadata
+
+# Build, dev and test dependencies
+, ipython
 , pytestCheckHook
-, pythonOlder
-, recommonmark
-, six
-, sphinx
-, toolz
-, typing ? null
 , vega_datasets
+, black
+, sphinx
 }:
 
 buildPythonPackage rec {
   pname = "altair";
-  version = "4.2.2";
-  disabled = isPy27;
+  version = "5.0.1";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-OTmaJnxJsw0QLBBBHmerJjdBVqhLGuufzRUUBCm6ScU=";
+  # We use fetch from Github because the Pypi is missing the tests
+  src = fetchFromGitHub {
+    owner = "altair-viz";
+    repo = "altair";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-7bTrfryu4oaodVGNFNlVk9vXmDA5/9ahvCmvUGzZ5OQ=";
   };
 
   propagatedBuildInputs = [
-    entrypoints
+    jinja2
     jsonschema
     numpy
     pandas
-    six
     toolz
-    jinja2
-  ] ++ lib.optionals (pythonOlder "3.5") [ typing ];
+  ] ++ lib.optionals (pythonOlder "3.8") [ importlib-metadata ]
+    ++ lib.optionals (pythonOlder "3.11") [ typing-extensions ];
 
   nativeCheckInputs = [
-    glibcLocales
+    hatchling
+    black
     ipython
-    pytestCheckHook
-    recommonmark
     sphinx
     vega_datasets
+    pytestCheckHook
   ];
 
   pythonImportsCheck = [ "altair" ];
 
-  # avoid examples directory, which fetches web resources
-  preCheck = ''
-    cd altair/tests
-  '';
+  disabledTestPaths = [
+      "tests/test_examples.py" # Disabled because it uses network
+      "tests/vegalite/v5/test_api.py" # TODO: Disabled because of missing altair_viewer package
```

Comment:
```suggestion
      # Disabled because it requires internet connectivity
      "tests/test_examples.py" 
      # TODO: Disabled because of missing altair_viewer package
      "tests/vegalite/v5/test_api.py"
```